sql/core/src/test/scala/org/apache/spark/sql/execution/WholeStageCodegenSuite.scala

```diff
@@ -70,4 +70,13 @@ class WholeStageCodegenSuite extends SparkPlanTest with SharedSQLContext {
         p.asInstanceOf[WholeStageCodegen].child.isInstanceOf[Sort]).isDefined)
     assert(df.collect() === Array(Row(1), Row(2), Row(3)))
   }
+
+  test("typed filter should be included in WholeStageCodegen") {
+    val ds = sqlContext.range(10).filter(_ % 2 == 0)
```

Comment:
this is a risky thing because as soon as we introduce https://issues.apache.org/jira/browse/SPARK-14083 this test will be useless.

maybe we should introduce a config option now and then explicitly turn off that future optimization

cc @JoshRosen  - any good names for the config option?